crates/forge_domain/src/agent.rs

```diff
@@ -322,7 +317,10 @@ impl Agent {
         Ok(context.extend_tools(if tool_supported {
             tool_defs
         } else {
-            Vec::new()
+            tool_defs
+                .into_iter()
+                .filter(|tool| tool.name.as_str().starts_with(FORGE_STRIP))
+                .collect()
```

Comment:
If tools aren't supported then we can not add tools to the context. This file shouldn't have changed at all.